test/api/model/initial_snapshot_test.dart

```diff
@@ -62,4 +62,19 @@ void main() {
       'unread_message_ids': [11, 2, 3],
     })).throws<AssertionError>();
   });
+
+  test('UserSettings.emojiset handles various unknown values', () {
+    final unknownValues = ['apple', 'microsoft', 'facebook', ''];
+    for (final unknownValue in unknownValues) {
+      final settings = UserSettings.fromJson({
+        'twenty_four_hour_time': true,
+        'display_emoji_reaction_users': true,
+        'emojiset': unknownValue,
+        'presence_enabled': true,
+      });
```

Comment:
This `.fromJson` call will fail and need to be debugged each time we add a field to `UserSettings`.

Can you add a `userSettings` helper function in test/example_data.dart, on the pattern of similar helpers (like `streamMessage` or `initialSnapshot`), then I think this can just be

```suggestion
      final json = eg.userSettings().toJson()..['emojiset'] = unknownValue;
      final settings = UserSettings.fromJson(json);
```